plonky2/src/hash/hash_types.rs

```diff
@@ -35,24 +37,26 @@ impl<F: Field> HashOut<F> {
         elements[0..elements_in.len()].copy_from_slice(elements_in);
         Self { elements }
     }
+}
 
-    #[cfg(feature = "parallel")]
-    pub fn rand_from_rng<R: rand::Rng>(rng: &mut R) -> Self {
+impl<F> Sample for HashOut<F>
```

Comment:
Argh, it would be nice if we could just use the standard `Rng` methods, like

```rust
let elems: [F; 10] = rng.gen();
```

But seems like the design of `rand` makes this impossible if `F` is generic? Since the `Field` trait can't impose a requirement that `Standard` impl `Distribution<Self>`, and we can't have a blanket impl `Distribution<F>`.

Reply:
`rand` doesn't have the nicest way to do this, but it is possible to do it generically if `rand` were designed a bit differently. We can add this kind of convenience but it requires some significant boilerplate which I don't think is worth it at this point, perhaps in the future.